src/app/modules/payments/stripe.service.ts

```diff
@@ -556,12 +557,21 @@ export const handleStripeEvent = (
             if (balanceTransaction.type !== 'charge') return
 
             const charge = balanceTransaction.source as Stripe.Charge
-            const innerResult = getMetadataPaymentId(
+            const innerResult = await getMetadataPaymentId(
               charge.metadata,
             ).asyncAndThen((paymentId) => processStripeEvent(paymentId, event))
 
             // Reducer to keep errors around
-            result = result.andThen(() => innerResult)
+            if (innerResult && innerResult.isErr()) {
+              logger.error({
+                message:
+                  'Error when calling processStripeEvent while paging through balanceTransaction',
+                meta: { ...logMeta, chargeId: charge },
+                error: innerResult,
+              })
+              payoutProcessError = errAsync(innerResult.error)
```

Comment:
since autoPagingEach will be iterating the balance transaction, wouldn't there be a chance that payoutProcesError gets overwritten? In that case, should we throw an error here and let the error handler handles it, instead of initialising a separate error variable?

Reply:
> since autoPagingEach will be iterating the balance transaction, wouldn't there be a chance that payoutProcesError gets overwritten?

Yes. It's only capturing the last error, or rather, the existence of an error. It's not meant to keep track of how many errors.

> In that case, should we throw an error here and let the error handler handles it

Throwing would short-circuit the process early, I think there is merit to letting non-erroneous transaction complete. Wdyt?

> instead of initialising a separate error variable

In case this too implicit, the intent of creating a separate error variable is to escape the everlasting promise.

Reply:
> Throwing would short-circuit the process early, I think there is merit to letting non-erroneous transaction complete. Wdyt?

Yeah, that makes sense. But, we just need to bear in mind that since we are not remembering which transaction we have already processed, in the event of retry by stripe, our webhook log would have some duplicated events, unless we want to add a logic to remove duplication. Otherwise we have to always be aware of the possibility of transaction being processed more than once when we add further computation towards the event that we received. 

Also, should we use mapErr instead after asyncAndThen?

Reply:
Refactored to `mapErr` in ef187748

> our webhook log would have some duplicated events

We have an event ID check for duplicates in `processStripeEvent`

Reply:
yes, you are right, it's being checked. Thanks for pointing it out. LGTM!